src/ptc/start.clj

```diff
@@ -72,61 +73,51 @@
           (.commit session))))))
 
 (defn listen-and-consume-from-queue
-  "Listen to QUEUE on CONNECTION for messages,
-  and call (TASK! message) until it is false."
+  "Call TASK! on messages from QUEUE on CONNECTION until it returns false."
   [task! connection queue]
   (loop [counter 0]
     (if-let [peeked (peek-message connection queue)]
-      ; to avoid NPE on ednify
       (let [peeked (jms/ednify peeked)]
-        (do
-          (log/infof "Peeked message %s: %s" counter peeked)
-          (if (task! peeked connection)
-            (let [consumed (jms/ednify (consume connection queue))]
-              (log/infof "Task complete, consumed message %s" counter)
-              (if (not (misc/message-ids-equal? peeked consumed))
-                (log/warnf
-                 (str/join \space ["Messages differ:"
-                                   (with-out-str (pprint (data/diff peeked consumed)))])))
-              (recur (inc counter)))
-            ;; this is for testing, in production, the task!
-            ;; should always return `true` and this branch should
-            ;; never gets reached.
-            (do
-              (log/errorf
-               (str/join
-                \space ["Task returned nil/false,"
-                        "not consuming message %s and instead exiting"])
-               peeked)
-              peeked))))
+        (log/infof "Peeked message %s: %s" counter peeked)
+        (if (task! peeked connection)
+          (let [consumed (jms/ednify (consume connection queue))]
+            (log/infof "Task complete, consumed message %s" counter)
+            (if (not (jms/message-ids-equal? peeked consumed))
+              (log/warnf
+               (str/join \space ["Messages differ:"
+                                 (with-out-str
+                                   (pprint (data/diff peeked consumed)))])))
+            (recur (inc counter)))
+          (do (log/errorf "Cannot consume %s" peeked)
```

Comment:
Is there any other info we have at this stage re: the cause of the error?  (The old comment said that this was only reached in testing?)

Reply:
It's used only for testing.  See the `forwarding` test, and so on.